### PR TITLE
TEST: add linter for TL CUDA

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Add new GH actions tests for CUDA. New workflow will run clang-tidy linter for all CUDA components in UCC including TL/CUDA, EC/CUDA, MC/CUDA. Also fixed found warning and errors.
New implementation of ucc_assert is needed to suppress false positive warnings, i.e. telling compiler that variable value within certain range.

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #
-# Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (shimmybalsam) [inline src/Makefile.am:2]: why did you remove 2022?

**Review by shimmybalsam — APPROVED** [review]

**Review by vspetrov — APPROVED** [review]

**Review by bureddy — APPROVED** [review]
